### PR TITLE
Cleanup substring code

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/Features/Core/Portable/QuickInfo/CommonSemanticQuickInfoProvider.cs [hunk 1]
```diff
@@ -191,7 +191,7 @@ void AddSection(string kind, ImmutableArray<TaggedText> taggedParts)
                         var builder = ImmutableArray.CreateBuilder<TaggedText>();
                         builder.AddText(FeaturesResources.Awaited_task_returns_0.Substring(0, symbolIndex));
                         builder.AddRange(mainDescriptionTaggedParts);
-                        builder.AddText(FeaturesResources.Awaited_task_returns_0.Substring(symbolIndex + defaultSymbol.Length));
+                        builder.AddText(FeaturesResources.Awaited_task_returns_0[(symbolIndex + defaultSymbol).Length..]);
```
Comment (davidwengier) [inline src/Features/Core/Portable/QuickInfo/CommonSemanticQuickInfoProvider.cs:194]: Those parentheses are not right. Bug in the fixer, or manual change?

**Review by davidwengier — APPROVED** [review]

File: src/Features/Core/Portable/QuickInfo/CommonSemanticQuickInfoProvider.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/QuickInfo/CommonSemanticQuickInfoProvider.cs:194]: manual.  good catch.  =-0   

we've fixed this in the fixer.  but i'm running an older version.

Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/QuickInfo/CommonSemanticQuickInfoProvider.cs:194]:
```suggestion
                        builder.AddText(FeaturesResources.Awaited_task_returns_0[(symbolIndex + defaultSymbol.Length)..]);
```

**Review by 333fred — APPROVED** [review]
LGTM (commit 4)

**Pull request merged**